java/integTest/src/test/java/glide/SharedCommandTests.java

```diff
@@ -4586,6 +4603,145 @@ public void xpending_return_failures(BaseClient client) {
+                                                groupName,
+                                                consumer1,
+                                                1L,
+                                                new String[] {streamid_1},
+                                                claimOptions)
+                                        .get());
+        assertInstanceOf(RequestException.class, executionException.getCause());
+        assertTrue(executionException.getMessage().contains("NOGROUP"));
+
+        // Key exists, but it is not a stream
+        assertEquals(OK, client.set(stringkey, "bar").get());
+        executionException =
+                assertThrows(
+                        ExecutionException.class,
+                        () ->
+                                client
+                                        .xclaim(stringkey, groupName, consumer1, 1L, new String[] {streamid_1})
+                                        .get());
+        assertInstanceOf(RequestException.class, executionException.getCause());
+
+        executionException =
+                assertThrows(
+                        ExecutionException.class,
+                        () ->
+                                client
+                                        .xclaim(
+                                                stringkey,
+                                                groupName,
+                                                consumer1,
+                                                1L,
+                                                new String[] {streamid_1},
+                                                claimOptions)
+                                        .get());
+        assertInstanceOf(RequestException.class, executionException.getCause());
+
+        executionException =
+                assertThrows(
+                        ExecutionException.class,
+                        () ->
+                                client
+                                        .xclaimJustId(stringkey, groupName, consumer1, 1L, new String[] {streamid_1})
+                                        .get());
+        assertInstanceOf(RequestException.class, executionException.getCause());
+
+        executionException =
+                assertThrows(
+                        ExecutionException.class,
+                        () ->
+                                client
+                                        .xclaimJustId(
+                                                stringkey,
+                                                groupName,
+                                                consumer1,
+                                                1L,
+                                                new String[] {streamid_1},
+                                                claimOptions)
+                                        .get());
+        assertInstanceOf(RequestException.class, executionException.getCause());
+    }
+
```

Comment:
Do we need to test any of the following, or not necessary?
- non-existing group
- non-existing consumer
- negative and/or 0 value for min-idle
- negative retry count
- invalid last ID

Reply:
1. non-existing group is tested with a non-existent key (it's the same test). 
2. non-existing consumer creates the consumer. 
3. Turns out, negative values are accepted.  Since they are minimums, they just do the same as the defaults. 
4. negative values are... ignored :(
5. I've removed last ID, as it doesn't appear to be an external option and is not documented (not exactly sure... but I also wasn't able to get it work through various tests). 
Lettice.io also doesn't have Last ID available. 
6. I can add this test too...